redis/commands/graph/query_result.py

```diff
@@ -360,3 +360,166 @@ def cached_execution(self):
     @property
     def run_time_ms(self):
         return self._get_stat(INTERNAL_EXECUTION_TIME)
+
+
+class AsyncQueryResult(QueryResult):
+    def __init__(self):
+        pass
+
+    async def initialize(self, graph, response, profile=False):
+        self.graph = graph
+        self.header = []
+        self.result_set = []
+
+        # in case of an error an exception will be raised
+        self._check_for_errors(response)
+
+        if len(response) == 1:
+            self.parse_statistics(response[0])
+        elif profile:
+            self.parse_profile(response)
+        else:
+            # start by parsing statistics, matches the one we have
+            self.parse_statistics(response[-1])  # Last element.
+            await self.parse_results(response)
+
+        return self
+
+    async def parse_node(self, cell):
+        # Node ID (integer),
+        # [label string offset (integer)],
+        # [[name, value type, value] X N]
+
+        node_id = int(cell[0])
```

Comment:
Since the node_id isn't used until the end, perhaps place it closer to line 400 (the return). Readability nit.